renku/core/commands/status.py

```diff
@@ -17,88 +17,77 @@
 from renku.core.management.interface.client_dispatcher import IClientDispatcher
 from renku.core.models.provenance.activity import Activity, Usage
-from renku.core.utils import communication
+from renku.core.utils.git import get_object_hash
+from renku.core.utils.os import get_relative_path_to_cwd, get_relative_paths
 
 
-def _get_relative_path(client, path):
-    """Get a relative path to current working directory."""
-    return str((client.path / path).resolve().relative_to(os.getcwd()))
-
-
-def get_status():
+def get_status_command():
     """Show a status of the repository."""
     return Command().command(_get_status).require_migration().require_clean().with_database(write=False)
 
 
 @inject.autoparams()
-def _get_status(client_dispatcher: IClientDispatcher, activity_gateway: IActivityGateway):
+def _get_status(client_dispatcher: IClientDispatcher, activity_gateway: IActivityGateway, paths=None):
     client = client_dispatcher.current_client
 
-    latest_activities = activity_gateway.get_latest_activity_per_plan().values()
-
-    if client.has_external_files():
-        communication.warn(
-            "Changes in external files are not detected automatically. To update external files run "
-            "`renku dataset update -e`."
-        )
+    paths = paths or []
+    paths = get_relative_paths(base=client.path, paths=paths)
 
-    try:
-        communication.echo("On branch {0}".format(client.repo.active_branch))
-    except TypeError:
-        communication.warn("Git HEAD is detached!\n Please move back to your working branch to use renku\n")
+    latest_activities = activity_gateway.get_latest_activity_per_plan().values()
 
-    modified, deleted = _get_modified_paths(activities=latest_activities)
+    modified, deleted = _get_modified_paths(activities=latest_activities, repo=client.repo)
 
     if not modified and not deleted:
-        return None, None, None
+        return None, None, None, None
 
+    modified_inputs = set()
     stales = defaultdict(set)
+    stale_activities = defaultdict(set)
 
     for activity, usage in modified:
-        usage_path = _get_relative_path(client, usage.entity.path)
-        for generation in activity.generations:
-            generation_path = _get_relative_path(client, generation.entity.path)
-            stales[generation_path].add(usage_path)
-        downstream_activities = activity_gateway.get_downstream_activities(activity)
-        paths = [_get_relative_path(client, g.entity.path) for a in downstream_activities for g in a.generations]
-        for p in paths:
-            stales[p].add(usage_path)
+        activities = activity_gateway.get_downstream_activities(activity)
+        activities.add(activity)
 
-    modified = {_get_relative_path(client, v[1].entity.path) for v in modified}
+        usage_path = get_relative_path_to_cwd(client.path / usage.entity.path)
 
-    deleted = {_get_relative_path(client, d) for d in deleted}
+        for activity in activities:
```

Comment:
This was intentional because I included the parent activity in the list of its downstream to avoid code duplication for parent and its dependents. I renamed the parent activity to avoid getting linter warning.